### ISSUE TITLE
Google Safebrowsing can no longer be disabled on mobile Firefox

### ISSUE THREAD

**Issue by rsssxd** [fork-maintainers/iceraven-browser#208]
I am reposting this because I was shadowbanned from Reddit with no reason given by their Anti-Evil Operations Team for several days which means nobody saw this even after it was restored:

[No way to disable Phishing and Malware Protection for users who are suspicious of Google and leaking browsing history to Google.](https://github.com/mozilla-mobile/fenix/issues/14163)

Relevant content on the failure of anonymization:

[How safebrowsing fails to protect privacy](https://blog.trailofbits.com/2019/10/30/how-safe-browsing-fails-to-protect-user-privacy/)

[A Privacy Analysis of Google and Yandex Safe Browsing](https://hal.inria.fr/hal-01120186v4/document)

> Our experimental analysis estimates the rate of such collisions and shows that hashing and truncation fails to prevent re-identification when a user visits small-sized domains or certain URLs of larger domains. We further materialize this in the form of an algorithm that Google and Yandex could potentially employ to track users. We conclude this work by providing an analysis of the databases of Google and Yandex (Section 7). By crawling their databases, we detect a number of “suspicious” prefixes that we call orphans. Orphans trigger communication with the servers, but no full digest corresponds to them. We also observe several URLs which have multiples prefixes included in the blacklists. These provide concrete examples of URLs and domains that can be easily tracked by Google and Yandex.

https://github.com/mozilla-mobile/fenix/issues/14163#issuecomment-680291892

> Does it have the potential of sending full URL-s to Google? Yes, it does. From the page given by you:
"Otherwise, send the binary file's metadata to the remote application reputation server (browser.safebrowsing.downloads.remote.url) and block the download if the server indicates that the file isn't safe."
with the link on "metadata" leading to parts of code where there is setting in request properties of origin URL. If I read code correctly - https://dxr.mozilla.org/mozilla-central/source/toolkit/components/reputationservice/ApplicationReputation.cpp#1306 - it is stripped from query params, but full hostname + path ARE included in this case.

https://redd.it/j5i8h1
@r_privacy

#ff #firefox #Google

**Comment by CharmCityCrab** [comment]
Though I personally use Google safe browsing, it is at least a light form of third-party telemetry or something akin to it (Albeit potentially very useful to the end user), and given that, and that the core mission (or part of the core mission) of the Iceraven browser is to provide options, including some that Mozilla can't or won't offer in Firefox, I would support changing this into an on-by-default option that the user can switch to "disabled" status in the GUI settings (Perhaps with a warning about safety).

The first thing that might be worth checking is whether or not this has been disabled already in Iceraven via some of the proprietary telemetry removal patch and some of the related work that has been since then on issue #26.  Does anyone know?  Anyone want to look at the code or run a wireshark type program and see if Iceraven makes this particular connection to Google the same way the latest versions of Firefox for Android do?

The second thing that is worth mentioning is that the OP said that desktop Firefox has an option in about:config to disable this.  Iceraven has about:config.  So, it might be worth checking if the about:config entry is a) there in Iceraven and b) does anything (A lot of about:config entries in the Fenix family don't really work or work differently than on desktop.  That unreliability is an issue that developed upstream before the fork, not something Iceraven did.).

Thirdly, we may want to look at whether F-Droid Fennec (Which is actually a Fenix fork now) has done any work on this issue that we can absorb through the magic of open-source licensing.  It seems like the sort of thing they'd want to rip out, but I'm not sure if they know it's there, and F-Droid does allow telemetry as long as the code is open-source, and this *could* be (Not *is*, could be.  I have no idea.  It's just that Google does a lot of open-source stuff and Mozilla does a lot of open-source stuff...), so they may not have gotten rid of it.

Ideally, we'd have a UI option that would allow users to choose for themselves.

As an aside, I think it's very possible that Iceraven could develop into the Android browser of choice for the majority of people who use desktop Firefox forks that have no Android equivalents (In addition to all the other larger markets Iceraven is serving or could potentially serve).  Those folks tend to really hate Google, so this could be a bigger deal for those users than maybe it is for the initial group that came over.  It seems like if we can provide an option, we could cater to that without compromising the experience of those who prefer the security advantages Google safebrowsing offers even given it's downside.  If the default is to have it on, it keeps our browser secure "out of the box", and people who turn it off can be responsible for their own security because they will have made an affirmative choice to be.

**Comment by CharmCityCrab** [comment]
Also, because I live in the western hemisphere and thus tend to associate the safe browsing feature with Google, it just sunk in re-reading the initial post that Yandex is the provider of the safe-browsing feature for Firefox in some countries (Not here, fortunately).

I don't want to get political, but some people are going to be have real concerns over a browser that interacts with any large company based in Putin's Russia for any reason.  And I don't just mean "not liking it" as an abstract type of concern, I mean people in political opposition groups within Russia, people in border countries under siege or potentially under seige where Putin has or could obtain significant influence, etc. who could feel it increases their risk of being spied on by a hostile government even higher than it already is.  Whether the devs agree with those concerns or not, the concerns are going to exist and effect browser uptake and usage if indeed Iceraven is doing what Firefox is doing (It may not be, see above post) with no off-switch.

At minimum, if we don't provide an "off" switch for this entirely (And we should), we could at least provide a way for people to switch safe browsing providers as a milder alternative, and pick whichever one they dislike the least.

Actually, it occurs to me that if rooting through the ideas in my previous comment, as well as the other projects mentioned, doesn't yield useful code or useful coding ideas, we could see what Tor is doing with this, too, to see if any ideas or code can be found there.   By the end of typing the second paragraph of this post, I suddenly remembered Tor for some reason. ;)

**Comment by laurentlbm** [comment]
Iceraven gives access to `about:config`, so I think you can disable it from there.

**Comment by ghost** [comment]
Yeah from what I understand the only current way to disable it would be in about:config, which you can use in Iceraven. I guess I would support disabling it by default in Iceraven but would pretty much understand if it was decided not to be. In any case, hopefully there will be an option to disable it in vanilla Fenix in the future or an option added for Iceraven, if possible.

**Comment by interfect** [comment]
I think this might be a duplicate of #145 

If you toggle the about:config setting, does that actually turn it off? I seem to remember that in #145 it sounded like Safebrowsing is implemented in Android-Components outside of Geckoview and it thus didn't respect about:config.
